src/components/RedistributeCard.jsx

```diff
@@ -2,24 +2,38 @@ import React from "react";
 import {
   PlusOutlined,
   MinusOutlined,
-  PictureOutlined,
+  InfoCircleOutlined,
 } from "@ant-design/icons";
 
-const RedistributeCard = ({ weight, onIncrement, onDecrement, ...props }) => (
+const RedistributeCard = ({
+  onClickRedistributeCard,
+  onIncrement,
+  onDecrement,
+  ...props
+}) => (
   <div className="border-solid border-primary h-40 w-80 flex bg-white rounded-lg overflow-hidden shadow-md">
-    <div className="flex bg-primary w-1/3  items-center justify-center">
-      <PictureOutlined className="text-3xl text-white" />
+    <div
+      className="flex bg-primary w-1/3  items-center justify-center"
+      onClick={onClickRedistributeCard}
+    >
+      <InfoCircleOutlined className="text-3xl text-white" />
     </div>
     <div className="flex flex-col w-64 mt-2">
-      <div className="mx-3">
-        <span className="text-primary font-bold text-lg font-heading">Pumpkin Pie</span>
+      <div className="mx-3" onClick={onClickRedistributeCard}>
+        <span className="text-primary font-bold text-lg font-heading">
+          {props.name}
+        </span>
         <p className="text-xs text-black text-primary font-paragraph">
```

Comment:
`text-black` applies the same CSS property as `text-primary`, please only preserve one